projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -2780,3 +2792,89 @@ function wpcom_launchpad_get_latest_draft_id() {
 
 	return $cached_draft_id;
 }
+
+/**
+ * Checks if the current site primary domain is a WPCOM domain.
+ *
+ * @return bool Will return true if the primary domain is a WPCOM domain.
+ */
+function wpcom_launchpad_is_primary_domain_wpcom() {
+	if ( ! ( new Automattic\Jetpack\Status\Host() )->is_wpcom_platform() ) {
+		return false;
+	}
+
+	if ( ! class_exists( 'Domain_Mapping' ) || ! class_exists( 'WPCOM_Domain' ) ) {
+		return false;
+	}
+
+	$blog_id = get_current_blog_id();
+
+	// @phan-suppress-next-line PhanUndeclaredClassMethod -- Being checked before being called.
+	$primary_domain_mapping = Domain_Mapping::find_primary_by_blog_id( $blog_id );
+	$is_wpcom_domain        = true;
+
+	if ( null !== $primary_domain_mapping ) {
+		// @phan-suppress-next-line PhanUndeclaredClassMethod -- Being checked before being called.
+		$wpcom_domain = new WPCOM_Domain( $primary_domain_mapping->get_domain_name() );
+		// @phan-suppress-next-line PhanUndeclaredClassMethod
+		$is_wpcom_domain = $wpcom_domain->is_wpcom_tld();
+	}
+
+	return $is_wpcom_domain;
+}
+
+/**
+ * Check if the `check_ssl_status` task is complete.
+ *
+ * @return bool
+ */
+function wpcom_launchpad_is_ssl_task_completed() {
+	// Only run on WPCOM platform.
+	if ( ! ( new Automattic\Jetpack\Status\Host() )->is_wpcom_platform() ) {
+		return false;
+	}
+
+	$task_id = 'check_ssl_status';
+
+	// If the task is already complete, return true.
+	if ( wpcom_launchpad_is_task_option_completed( array( 'id' => $task_id ) ) ) {
+		return true;
+	}
+
+	if ( ! class_exists( 'Domain_Mapping' ) || ! class_exists( 'Domain_Certificate_Flag_Group' ) || ! class_exists( 'WPCOM\Container\DI' ) || ! class_exists( 'Domain_Certificate_Flags_Manager' ) ) {
+		return false;
+	}
+
+	$blog_id = get_current_blog_id();
+
+	// @phan-suppress-next-line PhanUndeclaredClassMethod -- Being checked before being called.
+	$primary_domain_mapping = Domain_Mapping::find_primary_by_blog_id( $blog_id );
+
+	// If the site doesn't have a primary domain mapping, the task is not complete.
+	// It's also worth noting that this condition will also be caught by the wpcom_launchpad_is_ssl_task_disabled
+	// function disabling the task.
+	if ( null === $primary_domain_mapping ) {
+		return false;
+	}
+	// This way of checking for the certificate was extracted from the get_ssl_status function in wp-content/rest-api-plugins/endpoints/domains-ssl.php
+	// @phan-suppress-next-line PhanUndeclaredClassMethod, PhanUndeclaredClassReference
+	$certificate_flag_manager = WPCOM\Container\DI::get( Domain_Certificate_Flags_Manager::class );
+	$certificate_flags        = $certificate_flag_manager->get_flags( $primary_domain_mapping->get_domain_name() );
```

Comment:
Do we need to be extra careful and make sure we are not checking with a WPCOM subdomain?

same as we did above

```
$wpcom_domain = new WPCOM_Domain( $primary_domain_mapping->get_domain_name() );
$is_wpcom_domain = $wpcom_domain->is_wpcom_tld();
```